ubi_manifest/app/api.py

```diff
@@ -10,12 +13,70 @@
 from .models import DepsolveItem, DepsolverResult, DepsolverResultItem, TaskState
 from .utils import get_items_for_depsolving, get_repo_classes
 
+REQUEST_TIMEOUT = 20
+
 router = APIRouter(prefix="/api/v1")
 
 
 @router.get("/status")
-def status() -> dict[str, str]:
-    return {"status": "OK"}
+def status() -> dict[str, Any]:
+    # Check workers
+    i = app.control.inspect()
+    workers_status = {
+        "availability": i.ping(),
+        "stats": i.stats(),
+        "registered_tasks": i.registered(),
+        "active_tasks": i.active(),
+        "scheduled_tasks": i.scheduled(),
+    }
+
+    # Check redis
+    redis_client = redis.from_url(app.conf.result_backend)
+    redis_status = {"Connected": redis_client.ping()}
+
+    # Check celery beat
+    heartbeat = redis_client.get("celery-beat-heartbeat")
+
+    if heartbeat:
+        last_heartbeat = datetime.fromisoformat(heartbeat.decode("utf-8"))
+        now = datetime.now()
+        time_from_last_heartbeat = now - last_heartbeat
+        if time_from_last_heartbeat < timedelta(minutes=2):
+            celery_beat_status = "OK"
+        else:
+            days = time_from_last_heartbeat.days
+            hours = time_from_last_heartbeat.seconds // 3600
+            minutes = (time_from_last_heartbeat.seconds - hours * 3600) // 60
+            celery_beat_status = (
+                f"Someting is wrong, last heartbeat task ran "
+                f"{days} days, {hours} hours and {minutes} minutes ago."
+            )
+    else:
+        celery_beat_status = "No heartbeat task ran yet. Wait a minute."
```

Comment:
I would probably  rather see something like `n/a` instead of long message. Or you can introduce a bit more complicated output - that will have short status field (ok, n/a, failed), and longer msg field.

```
Celery beat status:
   status: OK
   msg: Celery beat operable
```